core/src/test/java/com/gentics/mesh/core/webroot/WebRootEndpointTest.java

```diff
@@ -94,6 +103,46 @@ public void testReadBinaryNode() throws IOException {
 		assertNotNull(downloadResponse);
 	}
 
+	@Test
+	public void testReadS3BinaryNode() {
+		HibNode node = content("news_2015");
+
+		try (Tx tx = tx()) {
+			SchemaVersionModel schema = schemaContainer("content").getLatestVersion().getSchema();
+			S3BinaryFieldSchema s3BinaryFieldSchema = new S3BinaryFieldSchemaImpl();
+			s3BinaryFieldSchema.setName("s3");
+			s3BinaryFieldSchema.setLabel("Some label");
+			schema.addField(s3BinaryFieldSchema);
+			schema.setSegmentField("s3");
+			schemaContainer("content").getLatestVersion().setSchema(schema);
+			tx.success();
+		}
+		String uploadBody = "{\n" +
+				"\t\"language\":\"en\",\n" +
+				"\t\"version\":\"1.0\",\n" +
+				"\t\"filename\":\"test.jpg\"\n" +
+				"}";
+
+		// 2. Update the binary data
+		call(() -> client().updateNodeS3BinaryField(
+				PROJECT_NAME,
+				node.getUuid(),
+				"s3",
+				uploadBody
+		));
+		//uploading
+		File tempFile = createTempFile();
+		s3BinaryStorage().createBucket("test-bucket").blockingGet();
+		s3BinaryStorage().uploadFile("test-bucket", node.getUuid() + "/s3", tempFile, false).blockingGet();
+
+
+		// 3. Try to resolve the path
+		String path = "/News/2015/test.jpg";
+		MeshWebrootResponse response = call(() -> client().webroot(PROJECT_NAME, path, new VersioningParametersImpl().draft(),
+				new NodeParametersImpl().setResolveLinks(LinkType.FULL)));
+		assertNotNull(response);
```

Comment:
I would add here a ImageIO parser as [above](https://github.com/gentics/mesh/pull/1200/files#diff-499bbbb30d9775192ffa28412c1287800fa7e39cbc58cc2fddf16e9f71633c8cR173), to ensure we provide a valid binary stream.